jobs/MeterInstanceJob.js

```diff
@@ -0,0 +1,160 @@
+      const events = await this.getInstanceEvents()
+      logger.info('received events -> ', events)
+      let meterResponse = await this.meter(events)
+      return this.runSucceeded(meterResponse, job, done)
+    } catch (err) {
+      return this.runFailed(err, {}, job, done)
+    }
+  }
+  /* jshint ignore:end */
+
+  static getInstanceEvents() {
+    const options = {
+      resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.INSTANCE,
+      resourceType: CONST.APISERVER.RESOURCE_TYPES.SFEVENT,
+      query: {
+        labelSelector: `state in (${CONST.METER_STATE.TO_BE_METERED},${CONST.METER_STATE.FAILED})`
+      }
+    };
+    return apiServerClient.getResources(options);
+  }
+
+  /* jshint ignore:start */
+  static async meter(events) {
+    try {
+      logger.info(`Number of events to be metered in this run - ${events.length}`);
+      // Adding this comment as we are transitioning to async/await
+      // Note: The below Promise is not bluebird promise
+      const resultArray = await Promise.all(_.map(events, async (event) =>
+        await this.sendEvent(event)));
+      const successCount = resultArray.filter(r => r === true).length;
+      return {
+        totalEvents: events.length,
+        success: successCount,
+        failed: events.length - successCount,
+      }
+    } catch (err) {
+      logger.error(`Error occurred while metering all events : `, err);
+    }
+  }
+  /* jshint ignore:end */
+
+  static isServicePlanExcluded(options) {
+    const serviceId = _.get(options, 'service.service_guid');
+    const planId = _.get(options, 'service.plan_guid');
+    logger.info(`Checking if service ${serviceId}, plan: ${planId} is excluded`);
+    const plan = catalog.getPlan(planId);
+    return !plan.metered;
+  }
+
+  static enrichEvent(options) {
+    // Add region , service name and plan sku name of the event
+    const serviceId = _.get(options, 'service.service_guid');
+    const planId = _.get(options, 'service.plan_guid');
+    logger.info(`Enriching the metering event ${serviceId}, plan: ${planId}`);
+    options.service.id = catalog.getServiceName(serviceId);
+    options.service.plan = catalog.getPlanSKUFromPlanGUID(serviceId, planId);
+    options.service = _.omit(options.service, ['service_guid', 'plan_guid']);
+    options.consumer.region = config.metering.region;
+    return options;
+  }
+
+  static getPlanSKUFromPlanGUID(serviceGuid, planGuid) {
+    logger.info(`Getting Plan SKU for service ${serviceGuid}, plan: ${planGuid}`);
+    const service = _.chain(catalog.toJSON().services)
+      .map((s) => s.id === serviceGuid ? s : undefined)
+      .filter(s => s !== undefined)
+      .head()
+      .value();
+    return _
+      .chain(service.plans)
+      .map((p) => p.id === planGuid ? p.sku_name : undefined)
+      .filter(p => p !== undefined)
+      .head()
+      .value();
+  }
```

Comment:
I think you have moved it to [catalog](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/561/files#diff-55c660028f2fa662d1ac1a23277c1213R47). Then can you remove from here?